src/jdk.httpserver/share/classes/sun/net/httpserver/UnmodifiableHeaders.java

```diff
@@ -26,15 +26,21 @@
 package sun.net.httpserver;
 
 import java.util.*;
+import java.util.function.BiFunction;
 import com.sun.net.httpserver.*;
 
-class UnmodifiableHeaders extends Headers {
+public class UnmodifiableHeaders extends Headers {
 
-        Headers map;
+    private final Headers headers;
```

Comment:
Could you add a comment here that Headers is mutable, but no reference to it escapes.

Also, can you fix the indentation of the other existing methods.